src/output.rs

```diff
@@ -316,10 +316,35 @@ pub(crate) fn write_buffer(buffer: &[u8], f: &mut fmt::Formatter<'_>) -> fmt::Re
     if let Ok(buffer) = str::from_utf8(buffer) {
         write!(f, "{}", buffer)
     } else {
-        write!(f, "{:?}", buffer)
+        write_bytes(buffer, f)
```

Comment:
I think we should update `dump_buffer` to match
(ideally one would be implemented in terms of the other)

Reply:
Added in the new push. Although, I've replaced it with `DebugBytes` everywhere, maybe should have been changed the function to return `DebugBytes` instead.